pkg/ottl/ottlfuncs/func_trim.go

```diff
@@ -0,0 +1,45 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TrimArguments[K any] struct {
+	Target      ottl.StringGetter[K]
+	Replacement ottl.Optional[string]
+}
+
+func NewTrimFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Trim", &TrimArguments[K]{}, createTrimFunction[K])
+}
+
+func createTrimFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*TrimArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("TrimFactory args must be of type *TrimArguments[K]")
+	}
+
+	return trim(args.Target, args.Replacement), nil
+}
+
+func trim[K any](target ottl.StringGetter[K], replacement ottl.Optional[string]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		replacementString := replacement.Get()
+		val, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+		if replacementString == "" {
```

Comment:
If the `replacement` argument is optional, we could use the `.IsEmpty()` function to check whether a value was provided or not. 

I'd suggest changing it a bit making it similar to the `strings.Trim` function, meaning that the default whitespace replacement would only be used when `replacement.IsEmpty()`, and `!replacement.IsEmpty() && replacement.Get() == ""` would return the original string. 

Any thoughts @TylerHelmuth?